packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1800,6 +1800,9 @@ export class Cluster extends ClusterBase {
    * @param options options for creating a new nodegroup
    */
   public addNodegroupCapacity(id: string, options?: NodegroupOptions): Nodegroup {
+    if (options?.instanceTypes?.some(i => nodeTypeForInstanceType(i) === NodeType.INFERENTIA)) {
+      this.addNeuronDevicePlugin();
+    }
```

Comment:
```suggestion
    const hasInferentiaInstanceType = [
      options?.instanceType,
      ...options?.instanceTypes ?? [],
    ].some(i => i && nodeTypeForInstanceType(i) === NodeType.INFERENTIA);
    if (hasInferentiaInstanceType) {
      this.addNeuronDevicePlugin();
    }
```
Let's also check the deprecated `instanceType` attribute.
Can you please add unit tests as well?

Reply:
sure